### ISSUE TITLE
Wiz apprentice icon doesn't go away after borging/de-traitoring

### ISSUE THREAD

**Issue by Ausops** [tgstation/tgstation#9453]
what happened

> be admin
> get wiz apprentice
> ghost out, shove someone else in via ckey edit in vv
> go afk for a bit
> come back as a wizard impostor (protect the real wizard I assume)
> dick around, get lasered
> get borged
> red wizard apprentice icon still remains
> vekter detraitors me and the icon still remains

what I expect to happen

> icon goes away after borging

**Comment by Incoming5643** [comment]
Icons are mind attached and have to be manually removed, I think the same thing would happen to any other antag type with an icon unless there's special handling to prevent it (I think Revs have this with MMI's but beyond that I have no idea.)

Strictly speaking being a robot doesn't conflict with your protecting the wizard, you could have actually still greentexted as a robot as long as you didn't get laws or commands that told you you had to do otherwise.

So I mean I don't really know if I'd even consider this a bug, wizard brains are still magic no matter what you do with them (never forget the fireballing MMI)

**Comment by MrStonedOne** [comment]
Icons aren't technically mind attached, they are attached to the atom they are suppose to show over. But mob login/logout code generally moves icons about based on mind. Maybe this could use more checks, like maybe making a child of image called hudicon, and have children of that for each type, so that the subsystem could add checks better, when getting added to things.

/image/ is a datum, so it can be overridden as far as i know

**Comment by Lumipharon** [comment]
Why would expect to lose the icon? You're still an apprentice/cultist/whatever, just now you have an metal shell and have your actions limited by laws.

**Comment by Ausops** [comment]
I've been responding to ahelps that involve questions about antags and borging as borging having removed your antag objectives and status entirely. That's why I thought you'd lose the icon. 

The only exception to this might be staff-of-changed borg apprentices but that's another issue.

**Comment by Incoming5643** [comment]
Mechanically speaking getting borged does not remove your antag status nor does it remove your objectives. In fact if you need to "die a glorious death" and kill your target prior to getting borged, you can still get greentext.

**Comment by Lumipharon** [comment]
I know borging an antag removes them from the post game list of antags, which is in itself really dumb no matter how you look at it.

I also remember some talk/a pr about it literally removing antag, which I'm not sure got merged or not. I don't get borged very often.
This is dumb also, since being silicon does not prevent you from attempting to greentext, laws withstanding - which can be changed in the course of the round. You can also be deborged, so if it does remove antag, it really shouldn't.

**Comment by RemieRichards** [comment]
Being borged puts a note in your notes list that says basically "These objectives are null and void you've failed"

On NT I made procs to quickly remove antagonist status, I ported them here ages ago but did NOT make it so borgs actually removed all antag status like they did on NT, purely because there was a backlash or something relating to that at the time.

It makes more sense for borgs to mechanically remove antag than to stick a 1-2 sentence line saying "you done goofed" in your notes (that nobody reads)

**Comment by Lumipharon** [comment]
But it would mke even more sense then that to simply not remove antag at all.
It's still the same person, they just have a mtal body and laws that may or may not stop you from achieving your objective.

**Comment by RemieRichards** [comment]
Well no, because the message being there implies you have lost, the message is only there (and a comment there or that used to be there) Was a temporary (temporary means 3 years+ around here) solution until the coder figured out how to disable/remove all the objectives.

tl;dr you've lost, that's it, game over.

Which means removing your antag+objectives(+false hope) is more important.

Of course perhaps people no longer want borgs to remove antag? I suggest a poll that includes things like:
* Fail their objectives but leave them on the end screen
* Status quo
* Remove all their antag status mechanically, including probably an admin log just to say "this guy is no longer allowed to be a shit"

**Comment by Jordie0608** [comment]
Borged antags can get unborged and go do whatever they want again.

**Comment by RemieRichards** [comment]
@Jordie0608 are you suggesting that for the poll? or stating that as the current case because that's really not how it's handled mechanically.

**Comment by Jordie0608** [comment]
Currently with the exception of Rev and I think cult your status is retained and ruleswise you're an antag so long as you're in a borg body.

**Comment by RemieRichards** [comment]
Well, granted code is not the best place to draw evidence from but these should probably be removed then.

<B>Robotize transform:</B>
```O.mind.store_memory("In case you look at this after being borged, the objectives are only here until I find a way to make them not show up for you, as I can't simply delete them without screwing up round-end reporting. --NeoFite")``` - Yes NeoFite but his fucking name in IN GAME TEXT.

<B>Constructed borgs:</B>
```O.mind.store_memory("As a cyborg, any objectives listed here are null and void, and will be marked as failed. They are simply here for memory purposes.")```

**Comment by Jordie0608** [comment]
A traitor borg's objectives are marked as failed always (as far as I can tell), tested it as a borged antag with matyr and redtexted.

**Comment by Lumipharon** [comment]
Which is dumb. A silicon with an escape and murder onbjective can still do do bother quite easily, so long as it's laws allow (being asimov with a human kill target for instance, makes it difficult.). Hell, if your laws permit, you can ask to be deborged to achieve steal objectives and the like.

I'm not arguing that it doesn't current and intentionally removed antag, I'm saying there's no reason for it, that's why silicon laws exist.

**Comment by RemieRichards** [comment]
Silicon laws are shit, See the silicon rules we have to follow on top of laws.
If laws were perfect then sure, I'd agree with you, but they're garbage, which is why i believe it's better to explicitly remove their Antagonist status.

**Comment by Miauw62** [comment]
asimov is great and only really needs clarification for edge cases (lizards, hulks, wizards, &c), except a lot of people are assholes and like bending the rules and then throwing a shitstorm or validhunting like bolting EVA.

I have never, ever read the silicon policy and have gotten a silicon ban once for something that was blatantly against asimov and didnt have much to do with the policy (i was very pissed off at the time).

If you want to have another default lawset, this is EASILY possible, just contact MSO and tell him to flip the neccesary switches and edit the required file, I don't remember their names anymore.

Anyway, silicons should REALLY not just remove antag status, because people CAN be unborged (even if its really unlikely), If borgs should stay antag or be suspended from antag unless they're no longer a borg is, imo, a matter for the admins to discuss.

That said, letting a borg complete escape alive is dumb because it will make people stop borging people to make sure they dont get greentext.

**Comment by Lumipharon** [comment]
I wouldn't have a problem with borgs not being able to 'escape alive', since they're now a borg. But they can easily achieve objectives like stranding people on station, or depending on what lawset, murdering someone.

And even if they're asimov, they can loudly exclaim how they hope you die, even if they can't directly act on it, which is hilarious.
